src/unix/mod.rs

```diff
@@ -136,9 +136,18 @@ s! {
     }
 
     pub struct hostent {
```

Comment:
`hostent` and `linger` seem to have pretty different definitions between Cygwin and the rest of Unix. Would it be possible to move the existing structures into a `cfg(not(target_os = "cygwin"))` block, then redefine them in `cygwin/mod.rs` instead?